pkg/discovery/repository/kube_cluster.go

```diff
@@ -192,6 +196,59 @@ func (summary *ClusterSummary) computePodToServiceMap() {
 	}
 }
 
+// Find and cache all static pods modeled as daemons for discover workers
+// to set daemon flag for pod entities. Returns mapping of pod UID to flag.
+func (summary *ClusterSummary) computeStaticPodToDaemonMap() {
+	glog.V(2).Info("creating mapping for static pods modeled as daemons.")
+	// static pods are only exposed through the API via mirror pods
+	// https://kubernetes.io/docs/reference/glossary/?all=true#term-mirror-pod
+	mirrorPods := util.GetMirrorPods(summary.Pods)
+	// common name prefix is our assumed conventions for showing intent of creating
+	// daemon sets with static pods
+	prefixToNodeNames := util.GetMirrorPodPrefixToNodeNames(mirrorPods)
+	nodePoolToNodes := util.MapNodePoolToNodes(summary.Nodes, summary.MachineSetToNodesMap)
+	// additionally includes every node as its own node group with a generated key to ensure uniqueness
+	nodePoolToNodes["cluster-"+base64.StdEncoding.EncodeToString([]byte(summary.Name))] = summary.Nodes
+	staticPodToDaemonMap := make(map[string]bool)
+	// cache of static pod prefix to flag for quicker look up
+	prefixToDaemon := make(map[string]bool)
+	daemonCount := 0
+	for _, pod := range mirrorPods {
+		prefix, ok := util.GetMirrorPodPrefix(pod)
+		nodeNames := prefixToNodeNames[prefix]
```

Comment:
Could `nodeNames` ever be `nil`, or the `prefix` is guaranteed to exist in the map?

Reply:
There is no chance, but I can write some defensive code.